src/analysis/asymptotics/negligible.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2021 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import analysis.asymptotics.asymptotics
+import analysis.asymptotics.specific_asymptotics
+import data.polynomial.eval
+
+/-!
+# Negligible Functions
+
+This file defines a predicate `negligible f` on functions `f` from `ℕ` to a normed field `𝕜`.
+
+The main theorem is `negligible_polynomial_mul` that says the product of a polynomial
+  and a negligible function is still a negligible function
+-/
+
+namespace asymptotics
+
+/-- Definition of negligible functions over an arbitrary `normed_field`.
+  Note that the second function always has type `ℕ → ℝ`, which generally gives better lemmas. -/
```

Comment:
The docstring should explain the definition to a reader who (a) doesn't know the definition; (b) does not see the code (e.g., reads API docs).

Reply:
BTW, why do you want the codomain to be a normed ring? What's wrong with a normed space?

Reply:
I generalized the definition to any space with `has_norm` (assuming that's what you were suggesting here), and relaxed the assumptions on some of the theorems